### PR TITLE
Initialise "success" variable to fix warnings

### PR DESCRIPTION
**Pull request by rishabhkumar296**

### CODE REVIEW

**Comment by rishabhkumar296** [comment]
Any suggestions why the travis-ci build fails.

**Comment by rcurtin** [comment]
Don't worry about the travis failure, it is not related to your fix here.

File: src/mlpack/core/tree/rectangle_tree/r_plus_tree_descent_heuristic_impl.hpp [hunk 1]
```diff
@@ -43,8 +43,6 @@ size_t RPlusTreeDescentHeuristic::ChooseDescentNode(TreeType* node,
         node->Child(bestIndex).Bound();
     bound |= node->Dataset().col(point);
 
-    success = true;
-
```
Comment (rcurtin) [inline src/mlpack/core/tree/rectangle_tree/r_plus_tree_descent_heuristic_impl.hpp]: The change that you've made means that during this loop, `success` is not reinitialized to `true` each iteration.  So this introduces a bug.

**Comment by rcurtin** [comment]
I think you force pushed, so I did not see that there was a change.  This looks better, thanks for the contribution.  If you'd like to add your name to the list of contributors (even though this is a small contribution!) to `src/mlpack/core.hpp` and `COPYRIGHT.txt`, I'll merge that in too.

**Pull request merged**